### PR TITLE
Add missing dependencies

### PR DESCRIPTION
**Pull request by ben-grande**
- lib(ayatana-)appindictor-gtk3: necessary for notifications icons
- libnotify-bin: necessary for notify-send
- libnotify4/libnotify: necessary to register notifications
- dunst/notification-daemon: necessary to server notifications

For: QubesOS/qubes-issues#6426
Fixes: QubesOS/qubes-issues#8453
Fixes: QubesOS/qubes-issues#7991

### CODE REVIEW

File: rpm_spec/qubes-video-companion.spec.in [hunk 1]
```diff
@@ -37,6 +37,9 @@ BuildArch:      noarch
 Requires:       gstreamer1-plugins-good
 Requires:       python3
 Requires:       v4l-utils
+Requires:       libayatana-appindicator-gtk3
+Requires:       (dunst or notification-daemon)
```
Comment (DemiMarie) [inline rpm_spec/qubes-video-companion.spec.in:41]: Why this particular combination of `Requires`?  `xfce4-notifyd` provides `desktop-notification-daemon` on at least Fedora 39.

Comment (ben-grande) [reply rpm_spec/qubes-video-companion.spec.in:41]: The first reason is `dunst` with `libnotify` is very lightweight, the second reason is that I didn't know about `xfce4-notifyd`. Should I use `xfce4-notifyd` them?

Comment (DemiMarie) [reply rpm_spec/qubes-video-companion.spec.in:41]: I suggest using the `desktop-notification-daemon` meta-package on Fedora, and whichever meta-package Debian uses (if any).

File: rpm_spec/qubes-video-companion-dom0.spec.in [hunk 2]
```diff
@@ -19,6 +19,11 @@ BuildRequires:  python3-setuptools
 BuildRequires:  python3-devel
 
 Requires:       gstreamer1-plugins-good
+Requires:       v4l-utils
+## Fedora 37 doesn't have libayatana-appindicator-gtk3
+Requires:       libappindicator-gtk3
+Requires:       notification-daemon
```
Comment (marmarek) [inline rpm_spec/qubes-video-companion-dom0.spec.in:25]: The virtual name on Fedora is `desktop-notification-daemon`, bare `notification-daemon` package on Fedora is the GNOME's one...

**Review by marmarek — CHANGES_REQUESTED** [review]

File: rpm_spec/qubes-video-companion-dom0.spec.in (hunk 2 above)
Comment (ben-grande) [reply rpm_spec/qubes-video-companion-dom0.spec.in]: Done

**Comment by marmarek** [comment]
This won't fly on R4.1:
```
  - nothing provides libayatana-appindicator-gtk3 needed by qubes-video-companion-sender-1.0.7-1.1.fc32.noarch from @commandline
```
Does it mean the sender is incompatible with dom0 in R4.1 anyway? If that's the case, maybe don't build it (change `+=` in `Makefile.builder` with just `:=`) ?

**Comment by ben-grande** [comment]
[If Ayatana is not present, it falls back to plain AppIndicator](https://github.com/QubesOS/qubes-video-companion/blob/main/sender/tray_icon.py#L24-L32).

Also Dom0 with Fedora 37 does not have Ayatana also and I have been screensharing Dom0.

See my comment in `rpm_spec/qubes-video-companion-dom0.spec.in`:
```
## Fedora 37 doesn't have libayatana-appindicator-gtk3
Requires:       libappindicator-gtk3
```
Which is just a comment I left to remember to later fix the package name on the next Dom0 major upgrade.

**Comment by marmarek** [comment]
Ok, but this means it's really a dependency of the sender subpackage, not dom0 one, no? Can you move it to the right place but make it conditional (`%if 0%{?fedora} <= 37`)? You can use %else too.

**Comment by ben-grande** [comment]
> Ok, but this means it's really a dependency of the sender subpackage, not dom0 one, no? Can you move it to the right place but make it conditional (`%if 0%{?fedora} <= 37`)? You can use %else too.

Added the if statement to `rpm_spec/qubes-video-companion-dom0.spec.in`.

There is no sender subpackage in the dom0 spec, only in the domU spec. The dom0 spec does not try to install a subpackage. I am therefore unsure what you mean of moving that requires to the correct place, as Dom0 spec does not have a subpackage.

**Comment by marmarek** [comment]
There is some confusion in the package split - dom0 gets both -dom0 and -sender packages built for it, and both contain relevant qrexec services and all sender files. It should be either only -dom0 being built for dom0, or the -dom0 one should have only dom0-only files (default policy) and depend on -sender.

**Comment by ben-grande** [comment]
> There is some confusion in the package split - dom0 gets both -dom0 and -sender packages built for it, and both contain relevant qrexec services and all sender files. It should be either only -dom0 being built for dom0, or the -dom0 one should have only dom0-only files (default policy) and depend on -sender.

You are right, changed the `Makefile.builder` dom0 section to use `:=`.

**Review by marmarek — APPROVED** [review]

**Pull request merged**